backend/src/nodes/impl/rembg/session_factory.py

```diff
@@ -0,0 +1,19 @@
+from typing import Type
+
+import onnxruntime as ort
+
+from ...impl.onnx.utils import get_input_shape
+from .session_base import BaseSession
+from .session_cloth import ClothSession
+from .session_simple import SimpleSession
+
+
+def new_session(session: ort.InferenceSession) -> BaseSession:
```

Comment:
Please explain what sessions are here.

Reply:
From what I can tell, what rembg does is wrap the onnx runtime session in a custom session class. There are a couple of different classes, depending on the model. They do some extra pre- and post-processing on the input/output data.

Reply:
Wdym "from what I can tell?" This function references one of our functions, so how can it be from rembg?

Reply:
I thought you were asking about the Session classes in general.

Here's the original version: https://github.com/danielgatis/rembg/blob/main/rembg/session_factory.py

session_factory.py exists to acquire the models, determine what kind of session wrapper that model needs, and set of the InferenceSession. I modified it heavily to work within chaiNNer.

1. We don't want to determine the models by filename, so I replaced that check with one that uses one of our functions to reference a unique characteristic of the cloth model vs the others.
2.  We aren't downloading the models, so I removed all that code.
3. We already have our own method of creating and caching InferenceSessions, so I passed it in instead of creating a new one.